File: grades.cpp
#include <iostream>
#include <string>

using namespace std;

/*
 * Asks for score, in fraction form
 * Asks for weight, in percentage form
 */

int main() {

    double inp;
    bool loop = true;
    while (loop) {
        cout << "Enter score as fraction ('n' to stop)" << '\n';
        cin >> inp;

        if (to_string(inp) == "n" || to_string(inp) == "N") {
            loop = false;
        }
        else {
            cout << inp << '\n';
        }
    }
    return 0;
}

File: grades.py
grade = 0
total_weight = 0
while (True):
    inp = float(input('Enter score as fraction (\'n\' to stop)'))
    if (inp == 'idk' or inp == 'n'):
        break
    weight = float(input('weight? (as a percentage)'))
    grade += inp * (weight / 100)
    total_weight += (weight / 100)

if abs(total_weight - 100) < 0.1:
    print(grade)
else:
    leftover_weight = 100 - total_weight
    print('leftover weight:', leftover_weight)
    currGrade = input('what is your current overall final grade? (in percentage form)')
    print('final exam grade:', (currGrade / 100 - grade) / leftover_weight)
